src/charm.py

```diff
@@ -1154,6 +1154,9 @@ def _on_update_status(self, _) -> None:
 
         self._set_primary_status_message()
 
+        # Restart topology observer if it is gone
+        self._observer.start_observer()
+
```

Comment:
We should most probably consider something like systemd timer unit in the future.